classlib/src/main/java/org/teavm/classlib/java/util/concurrent/atomic/TAtomicReference.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright 2018 martin below.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util.concurrent.atomic;
+
+import java.io.Serializable;
+import org.teavm.classlib.java.util.TObjects;
+
+/**
+ * @author martin below
+ */
+public class TAtomicReference<T> implements Serializable {
+    private T value;
+
+    public final boolean compareAndSet(T expect, final T update) {
+        if (!TObjects.equals(value, expect)) {
```

Comment:
According to [documentation](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/atomic/AtomicReference.html#compareAndSet(V,%20V)), referential comparison should be there.

Reply:
Also, this code won't work property in multi-threaded environment, see my notes to `TAtomicLong`.